src/pokeAPI.ts

```diff
@@ -11,12 +11,39 @@ export namespace PokeAPI {
       return await this.get(`https://pokeapi.co/api/v2/type/${type}/`);
     }
 
+    /**
+     * Retrieves stats about a given Pokemon
+     * @param pokemon The Pokemon to look up
+     * @returns The Pokemon's stats
+     */
+    async getPokemonStats(pokemon: string): Promise<PokemonStats> {
+      const data = await this.get(`https://pokeapi.co/api/v2/pokemon/${pokemon}`);
+      if (data) {
+        return {
+          name: data.name,
+          stats: data.stats.map((stat: any) => ({
+            name: stat.stat.name,
+            base_stat: stat.base_stat,
+            effort: stat.effort,
+          })),
+        };
+      } else {
+        console.log('no data found for', pokemon);
+        throw new Error(`No data found for ${pokemon}`);
+      }
+    }
+
     private cache: { [url: string]: any } = {};
 
     private async get(url: string) {
       if (this.cache[url] === undefined) {
-        const response = await axios.get(url);
-        this.cache[url] = response.data;
+        try{
+          const response = await axios.get(url);
+          this.cache[url] = response.data;
+        } catch (e) {
+          console.error('failed to fetch', url, e)
+          return null;
+        }
```

Comment:
Catching the error and returning `null` here is really broad. Returning `null` in the case of 404 might be OK, but I'd rather let errors bubble up and be caught on the highest level.